src/month_delta.rs

```diff
@@ -0,0 +1,31 @@
+use polars::prelude::*;
+
+pub(crate) fn impl_month_delta(start_dates: &Series, end_dates: &Series) -> PolarsResult<Series> {
+    let start_dates = start_dates.date()?;
+    let end_dates = end_dates.date()?;
+
+    if (start_dates.dtype() != &DataType::Date) || (end_dates.dtype() != &DataType::Date) {
+        polars_bail!(InvalidOperation: "polars_xdt.workday_count only works on Date type. Please cast to Date first.");
+    }
+
+    let month_diff: Int32Chunked = start_dates
+        .as_date_iter()
+        .zip(end_dates.as_date_iter())
+        .map(|(s_arr, e_arr)| {
+            s_arr.zip(e_arr).map(|(left, right)| {
+                let duration_delta = right.signed_duration_since(left);
+                let days_delta = duration_delta.num_days() as f32;
+                let year_fraction = days_delta / 365.;
+                let month_diff = year_fraction * 12.;
```

Comment:
this looks way too imprecise, sorry, we're going to need a robust solution in order to be able to merge

for example, what does this function give at the number of months between 2022-02-01 and 2022-03-01? I'd expect the answer to be 1, is that what this gives?

Reply:
Constructed a new simpler logic that will take this and many other case scenarios in consideration. 

Added more test scenarios.